client/ayon_traypublisher/plugins/create/create_editorial_advanced.py

```diff
@@ -0,0 +1,1341 @@
+    label = "Audio product"
+
+
+class EditorialModelInstanceCreator(EditorialClipInstanceCreatorBase):
+    """Model product type class
+
+    Model representation instance.
+    """
+    identifier = "editorial_model_advanced"
+    product_type = "model"
+    label = "Model product"
+
+
+class EditorialCameraInstanceCreator(EditorialClipInstanceCreatorBase):
+    """Camera product type class
+    Camera representation instance.
+    """
+    identifier = "editorial_camera_advanced"
+    product_type = "camera"
+    label = "Camera product"
+
+
+class EditorialWorkfileInstanceCreator(EditorialClipInstanceCreatorBase):
+    """Workfile product type class
+
+    Workfile representation instance.
+    """
+    identifier = "editorial_workfile_advanced"
+    product_type = "workfile"
+    label = "Workfile product"
+
+
+class EditorialAdvancedCreator(TrayPublishCreator):
+    """Advanced Editorial creator class
+
+    Advanced editorial workflow creator. This creator will process imput
+    editorial file and match its clips to files in folder.
+
+    Args:
+        TrayPublishCreator (Creator): Tray publisher plugin class
+    """
+    enabled = True
+    label = "Editorial Advanced"
+    product_type = "editorial"
+    identifier = "editorial_advanced"
+    default_variants = [
+        "main"
+    ]
+    description = "Editorial files to generate shots."
+    detailed_description = """
+Supporting publishing new shots to project
+or updating already created. Publishing will create OTIO file.
+"""
+    icon = "fa.file"
+    product_type_presets = []
+
+    def __init__(self, *args, **kwargs):
+        self._shot_metadata_solver = ShotMetadataSolver(self.log)
+        super(EditorialAdvancedCreator, self).__init__(*args, **kwargs)
+
+    def apply_settings(self, project_settings):
+        editorial_creators = deepcopy(
+            project_settings["traypublisher"]["editorial_creators"]
+        )
+        creator_settings = editorial_creators.get(self.identifier)
+
+        self.enabled = creator_settings.get("enabled", True)
+
+        self._shot_metadata_solver.update_data(
+            creator_settings["clip_name_tokenizer"],
+            creator_settings["shot_rename"],
+            creator_settings["shot_hierarchy"],
+            creator_settings["shot_add_tasks"]
+        )
+        self.product_type_presets = creator_settings[
+            "product_type_advanced_presets"]
+        if default_variants := creator_settings.get("default_variants"):
+            self.default_variants = default_variants
+
+    def create(self, product_name, instance_data, pre_create_data):
+        allowed_product_type_presets = self._get_allowed_product_type_presets(
+            pre_create_data)
+
+        clip_instance_properties = {
+            k: v
+            for k, v in pre_create_data.items()
+            if k != "sequence_filepath_data"
+            if k != "folder_path_data"
+            if k not in self.get_product_presets_with_names()
+        }
```

Comment:
That is a lot of calls `get_product_presets_with_names` method.
```suggestion
        ignored_keys = set(self.get_product_presets_with_names())
        ignored_keys |= {"sequence_filepath_data", "folder_path_data"}
        clip_instance_properties = {
            k: v
            for k, v in pre_create_data.items()
            if k not in ignored_keys
        }
```

Prefered way would be to specify keys you want to keep, instead of ignore. Especially because you should expect that it is called from api and not from UI.